src/ModelParser/VisibilityAwarePropertyAccessGuesser.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Liip\MetadataParser\ModelParser;
+
+use Liip\MetadataParser\Exception\ParseException;
+use Liip\MetadataParser\Metadata\PropertyAccessor;
+use Liip\MetadataParser\ModelParser\RawMetadata\PropertyVariationMetadata;
+use Liip\MetadataParser\ModelParser\RawMetadata\RawClassMetadata;
+use ReflectionClass;
+use ReflectionException;
+
+class VisibilityAwarePropertyAccessGuesser implements ModelParserInterface
+{
+    public function parse(RawClassMetadata $classMetadata): void
+    {
+        try {
+            $reflClass = new ReflectionClass($classMetadata->getClassName());
+        } catch (ReflectionException $e) {
+            throw ParseException::classNotFound($classMetadata->getClassName(), $e);
+        }
+
+        foreach ($classMetadata->getPropertyCollections() as $propertyCollection) {
+            foreach ($propertyCollection->getVariations() as $variation) {
+                $name = $variation->getName();
+
+                if (!$reflClass->hasProperty($name) || $reflClass->getProperty($name)->isPublic()) {
+                    continue;
+                }
+
+                $currentAccessor = $variation->getAccessor();
+
+                if (!($currentAccessor->getSetterMethod() && $currentAccessor->hasSetterMethod())) {
+                    $variation->setAccessor(new PropertyAccessor(
+                        $currentAccessor->getGetterMethod() ?: $this->guessGetter($reflClass, $variation),
+                        $currentAccessor->getSetterMethod() ?: $this->guessSetter($reflClass, $variation),
```

Comment:
cool, so you only add guessed methods if they are not already defined from something else :+1: